src/components/FullscreenModal/hooks/useStickyFooter.ts

```diff
@@ -0,0 +1,61 @@
+import { useCallback, useEffect, useState } from 'react';
+import debounce from 'lodash.debounce';
+import { isNull } from 'ramda-adjunct';
+
+import { useStickyObserver } from './useStickyObserver';
+
+export const useStickyFooter = (
+  modalRef: React.MutableRefObject<HTMLElement>,
+  modalFooterRef: React.MutableRefObject<HTMLElement>,
+): { isFixed: boolean; shouldShowScrollToTopButton: boolean } => {
+  const [isFixed, setIsFixed] = useState(false);
+  const [
+    shouldShowScrollToTopButton,
+    setShouldShowScrollToTopButton,
+  ] = useState(false);
+
+  const showScrollToTopButton = useCallback(() => {
+    if (isNull(modalRef.current)) return;
+    const isScrollable =
+      modalRef.current.scrollHeight > modalRef.current.offsetHeight;
+    setShouldShowScrollToTopButton(isScrollable);
+    setIsFixed(false);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const resizeListener = debounce(showScrollToTopButton, 200);
+  useEffect(() => {
+    showScrollToTopButton();
+    window.addEventListener('resize', resizeListener);
+
+    return () => {
+      window.removeEventListener('resize', resizeListener);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const isInView = useCallback(() => {
+    if (isNull(modalRef.current) || isNull(modalFooterRef.current)) return;
+    const scrollOffset =
+      modalRef.current.scrollTop + modalRef.current.offsetHeight;
+    const contentHeight =
+      modalRef.current.scrollHeight - modalFooterRef.current.scrollHeight;
+
+    if (isFixed && scrollOffset >= contentHeight) {
+      setIsFixed(false);
+    } else if (
+      !isFixed &&
+      scrollOffset < contentHeight - contentHeight * 0.001
```

Comment:
magic const? 😄

Reply:
`contentHeight * 0.001` is the threshold which tells when the footer gets sticky, I can extract that into const to be more explicit

Reply:
Yep, that would be helpful